### PR TITLE
[A11y Bug] TFM remove role button from badges.

### PR DESCRIPTION
**Pull request by dannyjdev**
## Changes
* `role="button"` removed from badges.
* Added aria-label to badges for screen reader.
* Fixed screen reader not announcing popover content on focus/click.

## Addresses
https://github.com/nuget/engineering/issues/4352

### CODE REVIEW

File: src/NuGetGallery/Scripts/gallery/common.js [hunk 1]
```diff
@@ -481,6 +479,25 @@
...
+
+            // Windows Narrator does not announce popovers' content. See: https://github.com/twbs/bootstrap/issues/18618
+            // We can force Narrator to announce the popover's content by "flashing" the element's ARIA label.
+            popoverElementDom.ariaLabel = "";
+
+            setTimeout(function () {
+                popoverElement.popover('hide');
+
+                // We need to restore the element's original ARIA label.
+                // Wait 0.15 seconds for the popover to fade away first.
```
Comment (agr) [inline src/NuGetGallery/Scripts/gallery/common.js:494]: The comment says 0.15 seconds, but `popoverFadeTimeMS` is set to be `200` above.

**Review by agr — APPROVED** [review]

**Review by advay26 — APPROVED** [review]

File: src/NuGetGallery/Scripts/gallery/common.js [hunk 2]
```diff
@@ -481,6 +479,25 @@
                 popoverElement.popover('hide');
             }
         });
+
+        function popoverShowAndHide() {
+            popoverElement.popover('show');
+
+            // Windows Narrator does not announce popovers' content. See: https://github.com/twbs/bootstrap/issues/18618
+            // We can force Narrator to announce the popover's content by "flashing" the element's ARIA label.
+            popoverElementDom.ariaLabel = "";
```
Comment (advay26) [inline src/NuGetGallery/Scripts/gallery/common.js:488]: Why do have to set it to an empty value here to "flash" it? Would re-assigning the originalLabel value to the ariaLabel not work the same way here?

`popoverElementDom.ariaLabel = originalLabel;`

Comment (dannyjdev) [reply src/NuGetGallery/Scripts/gallery/common.js:488]: Narrator was not announcing aria-label when the content was visible. This piece of code copied from https://github.com/NuGet/NuGetGallery/blob/main/src/NuGetGallery/Scripts/gallery/page-display-package.js#L62.

File: src/NuGetGallery/Scripts/gallery/common.js (hunk 1 above)
Comment (dannyjdev) [reply src/NuGetGallery/Scripts/gallery/common.js:494]: This code was copied from here: https://github.com/NuGet/NuGetGallery/blob/main/src/NuGetGallery/Scripts/gallery/page-display-package.js#L71. I think is about waiting for the popover to fade after 0.15 seconds so it's safe to use 200 ms to assign the aria-label again.

**Pull request merged**